nsxt/resource_nsxt_ip_pool_allocation_ip_address_test.go

```diff
@@ -0,0 +1,140 @@
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNSXIpPoolAllocationIPAddressCheckDestroy(state, poolResourceName, resourceName)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNSXIpPoolAllocationIPAddressCreateTemplate(poolName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNSXIpPoolAllocationIPAddressExists(poolResourceName, resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "ip_pool_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "allocation_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccNSXIpPoolAllocationIPAddressExists(poolResourceName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		exists, err := checkAllocationIPAddressExists(state, poolResourceName, resourceName)
+		if err != nil {
+			return err
+		}
+		if !*exists {
+			return fmt.Errorf("Allocation %s in IP Pool %s is not existing", resourceName, poolResourceName)
+		}
+		return nil
+	}
+}
+
+func getIpPoolIdByResourceName(state *terraform.State, poolResourceName string) (string, error) {
+	rsPool, ok := state.RootModule().Resources[poolResourceName]
+	if !ok {
+		return "", fmt.Errorf("IP Pool resource %s not found in resources", getIpPoolName())
+	}
+
+	poolID := rsPool.Primary.ID
+	if poolID == "" {
+		return "", fmt.Errorf("IP Pool resource ID not set in resources ")
+	}
+	return poolID, nil
+}
+
+func checkAllocationIPAddressExists(state *terraform.State, poolResourceName string, resourceName string) (*bool, error) {
+	poolID, err := getIpPoolIdByResourceName(state, poolResourceName)
+	if err != nil {
+		return nil, err
+	}
+
+	rs, ok := state.RootModule().Resources[resourceName]
+	if !ok {
+		return nil, fmt.Errorf("IP Pool allocation_ip_address resource %s not found in resources", resourceName)
+	}
+
+	nsxClient := testAccProvider.Meta().(*nsxt.APIClient)
+	listResult, responseCode, err := nsxClient.PoolManagementApi.ListIpPoolAllocations(nsxClient.Context, poolID)
+	if err != nil {
+		return nil, fmt.Errorf("Error while retrieving allocations for IP Pool ID %s. Error: %v", poolID, err)
+	}
+
+	if responseCode.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Error while checking if allocations in IP Pool %s exists. HTTP return code was %d", poolID, responseCode.StatusCode)
+	}
+
+	exists := false
+	for _, allocationIpAddress := range listResult.Results {
+		if allocationIpAddress.AllocationId == rs.Primary.ID {
+			exists = true
+			break
+		}
+	}
+
+	return &exists, nil
+}
+
+func testAccNSXIpPoolAllocationIPAddressCheckDestroy(state *terraform.State, poolResourceName, resourceName string) error {
+	// PoolManagementApi.ListIpPoolAllocations() call does not return updated list within two minutes.
+	// Therefore the destroy check is only performed sporadically of test runs (per random) to avoid long test run
+	rnd100 := rand.NewSource(time.Now().UnixNano()).Int63() % 100
```

Comment:
I think we better test this every time, it takes more than an hour to run the tests today